src/plugins/memdump/memdump.cpp

```diff
@@ -753,6 +753,103 @@ static event_response_t write_virtual_memory_hook_cb(drakvuf_t drakvuf, drakvuf_
     return VMI_EVENT_RESPONSE_NONE;
 }
 
+static event_response_t create_remote_thread_hook_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info) {
+    // First check if NtCreateThreadEx syscall was invoked from CreateRemoteThread. 
+    // In such case the target process should differ from the caller.
+
+    // IN HANDLE ProcessHandle
+    addr_t target_process_handle = drakvuf_get_function_argument(drakvuf, info, 4);
+
+    vmi_pid_t target_process_pid;
+    if (!drakvuf_get_pid_from_handle(drakvuf, info, target_process_handle, &target_process_pid)) {
+        PRINT_DEBUG("[MEMDUMP] Failed to retrieve target process pid\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    addr_t caller_process = drakvuf_get_current_process(drakvuf, info);
+    if (!caller_process) {
+        PRINT_DEBUG("[MEMDUMP] Failed to retrieve caller process\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+    vmi_pid_t caller_process_pid;
+    if (!drakvuf_get_process_pid(drakvuf, caller_process, &caller_process_pid)) {
+        PRINT_DEBUG("[MEMDUMP] Failed to retrieve caller process pid\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    if (target_process_pid == caller_process_pid) {
+        // NtCreateThreadEx has not been invoked from CreateRemoteThread
+        // and so it's not suspicious enought to create dump.
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    // Now retrieve information about the segment to which the StartRoutine 
+    // points to. Double check if it's executable and if so – dump this segment.
+
+    // IN PVOID StartRoutine
+    addr_t start_routine = drakvuf_get_function_argument(drakvuf, info, 5);
+
+    // Retrieve target_process as start_routine points inside it's address space.
+    addr_t target_process;
+    if (!drakvuf_find_process(drakvuf, target_process_pid, nullptr, &target_process)) {
+        PRINT_DEBUG("[MEMDUMP] Failed to retrieve target_process\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    mmvad_info_t mmvad;
+    if (!drakvuf_find_mmvad(drakvuf, target_process, start_routine, &mmvad)) {
+        PRINT_DEBUG("[MEMDUMP] Failed to find mmvad\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
```

Comment:
Maybe replace `drakvuf_lock_and_get_vmi` with `vmi_lock_guard`? Then you don't have to remember about unlocking vmi when returning early.